kolibri/core/content/utils/resource_import.py

```diff
@@ -512,29 +515,75 @@ class ContentDownloadRequestResourceImportManager(RemoteChannelResourceImportMan
     def __init__(
         self,
         channel_id,
-        peer_id=None,
-        baseurl=None,
-        node_ids=None,
-        exclude_node_ids=None,
+        peer,
+        download_request,
         renderable_only=True,
         fail_on_error=False,
         content_dir=None,
-        download_request=None,
         timeout=transfer.Transfer.DEFAULT_TIMEOUT,
     ):
         super(ContentDownloadRequestResourceImportManager, self).__init__(
             channel_id,
-            peer_id=peer_id,
-            baseurl=baseurl,
-            node_ids=node_ids,
-            exclude_node_ids=exclude_node_ids,
+            peer_id=peer.id,
+            baseurl=peer.baseurl,
+            node_ids=[download_request.node_id],
+            exclude_node_ids=None,
             renderable_only=renderable_only,
             fail_on_error=fail_on_error,
             content_dir=content_dir,
             timeout=timeout,
         )
+        self.peer = peer
         self.download_request = download_request
 
+    def get_import_data(self):
+        return get_import_export_data(
+            self.channel_id,
+            self.node_ids,
+            self.exclude_node_ids,
+            False,
+            renderable_only=self.renderable_only,
+            all_thumbnails=self.all_thumbnails,
+            peer_id=self.peer_id,
+            filter_by_file_availability=False,
+        )
+
+    def run(self):
+        node = ContentNode.objects.get(pk=self.download_request.contentnode_id)
+        required_checksums = (
+            node.files()
+            .filter(supplementary=False)
+            .values_list("local_file_id", flat=True)
+        )
+        with NetworkClient.build_from_network_location(self.peer) as client:
+            response = client.post(
+                "api/public/v{version}/file_checksums/{channel_id}".format(
+                    version="1", channel_id=self.channel_id
+                ),
+                data=compress_string(
+                    bytes(json.dumps(list(required_checksums)).encode("utf-8"))
+                ),
+                headers={"content-type": "application/gzip"},
+            )
+            try:
+                response.raise_for_status()
```

Comment:
The network client would already call this with `client.post`, and would raise `NetworkLocationResponseFailure` instead